skactiveml/pool/regression/_information_gain_maximization.py

```diff
@@ -0,0 +1,501 @@
+        )
+
+        if mapping is None:
+            utilities = utilities_cand
+        else:
+            utilities = np.full(len(X), np.nan)
+            utilities[mapping] = utilities_cand
+
+        return simple_batch(
+            utilities,
+            self.random_state_,
+            batch_size=batch_size,
+            return_utilities=return_utilities,
+        )
+
+    def _kullback_leibler_divergence(
+        self, X_eval, X_cand, mapping, reg, X, y, sample_weight=None
+    ):
+        """Calculates the mutual information gain over the evaluation set if each
+        candidate sample where to be labeled.
+
+        Parameters
+        ----------
+        X_eval : array-like of shape (n_samples, n_features)
+            The samples where the information gain should be evaluated.
+        X_cand : array-like of shape (n_candidate_samples, n_features)
+            The candidate samples that determine the information gain.
+        mapping : array-like of shape (n_candidate_samples,) or None
+            A mapping between `X_cand` and `X` if it exists.
+        reg: ProbabilisticRegressor
+            Predicts the entropy, predicts values.
+        X : array-like of shape (n_samples, n_features)
+            Training data set, usually complete, i.e. including the labeled and
+            unlabeled samples.
+        y : array-like of shape (n_samples)
+            Labels of the training data set (possibly including unlabeled ones
+            indicated by `self.missing_label`).
+        sample_weight: array-like of shape (n_samples), optional (default=None)
+            Weights of training samples in `X`.
+
+        Returns
+        -------
+        query_indices : numpy.ndarray of shape (n_candidate_samples)
+            The expected information gain for each candidate sample.
+        """
+
+        def new_kl_divergence(idx, x_cand, y_pot):
+            reg_new = update_reg(
+                reg,
+                X,
+                y,
+                sample_weight=sample_weight,
+                y_update=y_pot,
+                idx_update=idx,
+                X_update=x_cand,
+                mapping=mapping,
+            )
+            entropy_post = np.sum(
+                reg_new.predict(X_eval, return_entropy=True)[1]
+            )
+            cross_ent = np.sum(
+                cross_entropy(
+                    X_eval,
+                    reg_new,
+                    reg,
+                    integration_dict=self.integration_dict_cross_entropy,
+                    random_state=self.random_state_,
+                )
+            )
+            return cross_ent - entropy_post
+
+        kl_div = conditional_expect(
+            X_cand,
+            new_kl_divergence,
+            reg,
+            random_state=self.random_state_,
+            include_idx=True,
+            include_x=True,
+            **self.integration_dict_target_val
+        )
+
+        return kl_div
+
+
+def cross_entropy(
```

Comment:
I wouldn't expect to import the `cross_entropy` from `pool.regression._information_gain_maximization` . Maybe you can find a more suitable location for this function?